tests/store/test_in_memory_backing_store.py

```diff
@@ -0,0 +1,195 @@
+    mock_user.colleagues[0].backing_store.return_only_changed_values = True
+    changed_values = mock_user.backing_store.enumerate_()
+    assert len(changed_values) == 1
+    assert  changed_values[0][0] == "colleagues" # Backingstore should detect manager property changed
+    nested_changed_values = mock_user.colleagues[0].backing_store.enumerate_()
+    assert len(nested_changed_values) == 1
+    assert nested_changed_values[0][0] == "business_phones"
+    
+def test_backing_store_embedded_in_model_by_updating_nested_backed_model_collection_property_with_extra_value_returns_changed_nested_properties():
+    mock_user = MockEntity(
+        id="84c747c1-d2c0-410d-ba50-fc23e0b4abbe",
+        colleagues=[MockEntity(
+            id="2f8c9b8c-8ea0-46b2-9c6d-660c9b7bf6af",
+            business_phones=["+1 234 567 891"]
+            )]
+    )
+    mock_user.backing_store.is_initialization_completed = mock_user.colleagues[0].backing_store.is_initialization_completed = True
+    mock_user.colleagues[0].business_phones.append("+9 876 543 219")
+    mock_user.backing_store.return_only_changed_values = True
+    mock_user.colleagues[0].backing_store.return_only_changed_values = True
+    changed_values = mock_user.backing_store.enumerate_()
+    assert len(changed_values) == 1
+    assert  changed_values[0][0] == "colleagues" # Backingstore should detect manager property changed
+    nested_changed_values = mock_user.colleagues[0].backing_store.enumerate_()
+    assert len(nested_changed_values) == 1
+    assert nested_changed_values[0][0] == "business_phones"
+    assert nested_changed_values[0][1] == (["+1 234 567 891", "+9 876 543 219"], 2)
+    
+def test_backing_store_embedded_in_model_by_updating_nested_backed_model_collection_property_with_extra_backed_model_returns_changed_nested_properties():
+    mock_user = MockEntity(
+        id="84c747c1-d2c0-410d-ba50-fc23e0b4abbe",
+        colleagues=[MockEntity(
+            id="2f8c9b8c-8ea0-46b2-9c6d-660c9b7bf6af",
+            business_phones=["+1 234 567 891"]
+            )]
+    )
+    mock_user.backing_store.is_initialization_completed = mock_user.colleagues[0].backing_store.is_initialization_completed = True
+    mock_user.colleagues.append(MockEntity(id="2fe22fe5-1132-42cf-90f9-1dc17e325a74"))
+    mock_user.backing_store.return_only_changed_values = True
+    mock_user.colleagues[0].backing_store.return_only_changed_values = True
+    changed_values = mock_user.backing_store.enumerate_()
+    assert len(changed_values) == 1
+    assert  changed_values[0][0] == "colleagues" # Backingstore should detect manager property changed
+    colleagues = changed_values[0][1]
+    assert len(colleagues) == 2
+    assert colleagues[0][0].id == "2f8c9b8c-8ea0-46b2-9c6d-660c9b7bf6af"
+    assert colleagues[0][1].id == "2fe22fe5-1132-42cf-90f9-1dc17e325a74"
+    nested_changed_values = mock_user.colleagues[0].backing_store.enumerate_()
+    assert len(nested_changed_values) == 0
+        
+        
```

Comment:
Related to above comment. Perhaps add a test where the mock_user "getter" is invoked & test whether it returns a value only if `return_only_changed_values = True`. (Model serialization relies on getters being called & they should return a value only if it has changed since initialization)

Reply:
This is the actual behavior(see previous reply). But I'll add the test.

Reply:
Tests added here dfd385f912aadbce8b3587ad162515a5f2baeb40